File: structured-programming/functions/calculator_functions.cpp
#include <iostream>

using namespace std;

double addNumbers(double firstNumber, double secondNumber) {
    return firstNumber + secondNumber;
}

double subtractNumbers(double firstNumber, double secondNumber) {
    return firstNumber - secondNumber;
}

double multiplyNumbers(double firstNumber, double secondNumber) {
    return firstNumber * secondNumber;
}

bool divideNumbers(double firstNumber, double secondNumber, double &result) {
    if (secondNumber == 0) {
        return false;
    }

    result = firstNumber / secondNumber;
    return true;
}

int main() {
    double firstNumber = 12;
    double secondNumber = 4;
    double divisionResult;

    cout << "Calculator Function Examples" << endl;
    cout << "First number: " << firstNumber << endl;
    cout << "Second number: " << secondNumber << endl;
    cout << "Addition: " << addNumbers(firstNumber, secondNumber) << endl;
    cout << "Subtraction: " << subtractNumbers(firstNumber, secondNumber) << endl;
    cout << "Multiplication: " << multiplyNumbers(firstNumber, secondNumber) << endl;

    if (divideNumbers(firstNumber, secondNumber, divisionResult)) {
        cout << "Division: " << divisionResult << endl;
    } else {
        cout << "Division failed because division by zero is not allowed." << endl;
    }

    return 0;
}

File: python-basics/functions/calculator_functions.py
def add_numbers(first_number, second_number):
    return first_number + second_number


def subtract_numbers(first_number, second_number):
    return first_number - second_number


def multiply_numbers(first_number, second_number):
    return first_number * second_number


def divide_numbers(first_number, second_number):
    if second_number == 0:
        return None
    return first_number / second_number


def run_calculator_examples():
    first_number = 12
    second_number = 4

    print("Calculator Function Examples")
    print("First number:", first_number)
    print("Second number:", second_number)
    print("Addition:", add_numbers(first_number, second_number))
    print("Subtraction:", subtract_numbers(first_number, second_number))
    print("Multiplication:", multiply_numbers(first_number, second_number))

    division_result = divide_numbers(first_number, second_number)
    if division_result is None:
        print("Division: cannot divide by zero")
    else:
        print("Division:", division_result)


if __name__ == "__main__":
    run_calculator_examples()
